### ISSUE TITLE
LSQ method

### ISSUE THREAD

**Issue by kanybekasanbekov** [blueardour/detectron2#2]
Hi @blueardour,
First of all, I would like to thank you for your amazing work. I came to this repository after reading this [paper](https://arxiv.org/pdf/2007.06919.pdf). I have a few questions about the implementation of LSQ. I have found some interesting(confusing) points while I have been going through the code and config files.
1. Any config file doesn't have an argument for `grad_scale`, so I assume it will be assigned to the default value `none`. That leads to `grad_factor=1.0`, so during fine-tuning (backpropagation) `clip_val` gradients aren't going to be scaled, right?
2. In every config file `boundary` was set to be equal to `1.0`. Does it mean that in the beginning fine-tuning, `clip_val` will start with the initial value `1.0` and learn the proper value later during training? Why didn't you use initialization methods like in the original LSQ paper? 
3. Have you done experiments with different initial values for arguments like `grad_scale`, `grad_type`, and etc.

**Comment by blueardour** [comment]
Hi, 

I have try a different kind of `grad_scale` in the `clip_val`, based on my experiments, there was never a performance improvement for this technique. Latter I thought about it and in my view, it had a simiar effect to change the learning rate of the `clip_val`. 

For initial value of the `clip_val`, the default value is 1.0 for most of the case. I also provided a function `init_from_warmup` when you open the `fm_stable/wt_stable` option. Thus, the `clip_val` is inited from first several iterations with the value of `correlate * max(tensor)`.  I found no clear final performance difference with the two methods on ResNet backbone. However, I indeed found the latter one might be more stable when training for shuffenet/mobilenet/ etc.

**Comment by kanybekasanbekov** [comment]
Can you help me to navigate to `init_from_warmup`? I couldn't find it, moreover, I don't see `fm_stable/wt_stable` options in a list of [arguments](https://github.com/aim-uofa/model-quantization/blob/master/task_cls.py) ...

Detectron2 folder in [Google Drive](https://drive.google.com/drive/folders/1opy6HwMzcNIw5Fh0IbyarLqplvxr22LE) is empty. Do you provide any weights(both fp and lsq) for Detectron2 models? If yes, where can I find them?

Thanks a lot.

**Comment by blueardour** [comment]
Hi, 

In current detectron2 branch,  we have an option to customized the `stable'
(around)[https://github.com/blueardour/detectron2/blob/073a8ad0f22178d1452df3c449f0e45b1ff50f8a/detectron2/config/defaults.py#L649]
 
If the stable is set, `self.init_based_on_warmup' will be called [https://github.com/aim-uofa/model-quantization/blob/67d4e313f09c84acd516f98789cdcdd39417f45d/models/quant.py#L446]
